### PR TITLE
scripted-diff: Regenerate key_io data deterministically 

### PR DESCRIPTION
**Pull request by maflcko**

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Requested by @Sjors here: https://github.com/bitcoin/bitcoin/pull/20861#pullrequestreview-596345213

**Comment by Sjors** [comment]
tACK fab2a4a, modulo the numbers in the `README` should be updated...

Also, I find it less tedious to run this from the root of the project:
```
PYTHONPATH=test/functional/test_framework contrib/testgen/gen_key_io_test_vectors.py valid 61 > src/test/data/key_io_valid.json
PYTHONPATH=test/functional/test_framework contrib/testgen/gen_key_io_test_vectors.py invalid 60 > src/test/data/key_io_invalid.json
git diff

```

**Comment by MarcoFalke** [comment]
Done

File: contrib/testgen/README.md [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 Utilities to generate test vectors for the data-driven Bitcoin tests.
 
-Usage:
+To use inside a scripted-diff (or just execute directly):
 
-    PYTHONPATH=../../test/functional/test_framework ./gen_key_io_test_vectors.py valid 50 > ../../src/test/data/key_io_valid.json
-    PYTHONPATH=../../test/functional/test_framework ./gen_key_io_test_vectors.py invalid 50 > ../../src/test/data/key_io_invalid.json
+    -BEGIN VERIFY SCRIPT-
```
Comment (Sjors) [inline contrib/testgen/README.md:7]: The `-BEGIN VERIFY SCRIPT-` bit is confusing.

**Comment by sipa** [comment]
This is generally a good idea, and we should do it. But this does still mean that changes to what is generated based on the RNG output will result in an entirely different output. E.g. adding an option to a random.choice() may cause more entropy to be extracted from the RNG, and thus cause all future data from it to be shifted. In theory this could be addresses using consistent hashing, but that's probably overkill.

Concept ACK

File: contrib/testgen/README.md (hunk 1 above)
Comment (MarcoFalke) [reply contrib/testgen/README.md]: fixed

**Comment by Sjors** [comment]
re-utACK fa21feb0b9c93fbc205d102510d3bb7a474431c5

File: contrib/testgen/gen_key_io_test_vectors.py [hunk 2]
```diff
@@ -4,14 +4,9 @@
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 '''
 Generate valid and invalid base58 address and private key test vectors.
-
-Usage:
-    PYTHONPATH=../../test/functional/test_framework ./gen_key_io_test_vectors.py valid 50 > ../../src/test/data/key_io_valid.json
-    PYTHONPATH=../../test/functional/test_framework ./gen_key_io_test_vectors.py invalid 50 > ../../src/test/data/key_io_invalid.json
 '''
 # 2012 Wladimir J. van der Laan
```
Comment (MarcoFalke) [inline contrib/testgen/gen_key_io_test_vectors.py:8]: @laanwj This says your name, which makes you qualified to review this? :grimacing:

**Comment by MarcoFalke** [comment]
Rebased

File: contrib/testgen/gen_key_io_test_vectors.py (hunk 2 above)
Comment (laanwj) [reply contrib/testgen/gen_key_io_test_vectors.py:8]: I definitely don't remember writing this :sweat:

Comment (MarcoFalke) [reply contrib/testgen/gen_key_io_test_vectors.py]: Your name is removed in current master, but you are still welcome to review

**Comment by Sjors** [comment]
ACK fa506add25cbe5efbbabca647f5378c4128cf945

I get the same JSON files when running the script. There are no duplicate addresses and they all change if you modify the `42` seed value (as expected).

**Comment by laanwj** [comment]
Tested ACK fa506add25cbe5efbbabca647f5378c4128cf945
Good idea to put it in the scripted-diff. I also verified locally that the output is the same.

**Pull request merged**